File: 374.Guess.Number.Higher.or.Lower.java
/* The guess API is defined in the parent class GuessGame.
   @param num, your guess
   @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
      int guess(int num); */

public class Solution extends GuessGame {
    public int guessNumber(int n) {
        int low = 1, high = n, mid, ret;
        while(low < high)
        {
            mid = low+(high-low)/2;
            ret = guess(mid);
            if(ret == -1)
            {
                high = mid;
            }
            else if (ret == 1)
            {
               low = mid + 1;
            }
            else
            {
                return mid;
            }
        }
        return high;
    }
}

File: 374. Guess Number Higher or Lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:
        l, r = 1, n
        while l <= r:
            m = (l + r) >> 1
            k = guess(m)
            if k == -1:
                r = m - 1
            elif k == 1:
                l = m + 1
            else:
                return m
        return l